presto-main/src/main/java/com/facebook/presto/sql/planner/ExpressionInterpreter.java

```diff
@@ -216,19 +218,29 @@ private ExpressionInterpreter(Expression expression, Metadata metadata, Session
     public Object evaluate(RecordCursor inputs)
     {
         checkState(!optimize, "evaluate(RecordCursor) not allowed for optimizer");
-        return visitor.process(expression, inputs);
+        return resolve(visitor.process(expression, inputs), metadata, session);
     }
 
     public Object evaluate(int position, Block... inputs)
     {
         checkState(!optimize, "evaluate(int, Block...) not allowed for optimizer");
-        return visitor.process(expression, new PagePositionContext(position, inputs));
+        return resolve(visitor.process(expression, new PagePositionContext(position, inputs)), metadata, session);
     }
 
     public Object optimize(SymbolResolver inputs)
     {
         checkState(optimize, "evaluate(SymbolResolver) not allowed for interpreter");
-        return visitor.process(expression, inputs);
+        return resolve(visitor.process(expression, inputs), metadata, session);
+    }
+
+    private static Object resolve(Object value, Metadata metadata, ConnectorSession session)
```

Comment:
Why do we need this? The visitor already calls `LiteralInterpreter.evaluate` when it encounters a `Literal` node.